src/pan3d/ui/contour.py

```diff
@@ -133,25 +128,25 @@ def __init__(self, source, update_rendering):
             )
 
     def update_from_source(self, source=None):
+        state = self.state
+        source = source or self._retrieve_source()
         if source is None:
-            source = self.source
+            return
 
         with self.state:
```

Comment:
If you write `with self.state as state:` you can remove line 131.

You can also rename it to something smaller if that helps (s).